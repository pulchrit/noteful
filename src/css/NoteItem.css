.note-item {
    padding: 15px;
    margin-bottom: 10px;
    background-color: lightgrey;
    border-radius: 2px;
}

.date-delete {
    display: flex;
    justify-content: space-between;
    align-items: center;
}

.date {
    font-size: .8em;
    width: 60%;
}

.delete-button {
    padding: 15px;
    border: 1px solid black;

}