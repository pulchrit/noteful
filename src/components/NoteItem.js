import React from 'react';
import {Link, withRouter} from 'react-router-dom';
import NotefulContext from './NotefulContext';
import "../css/NoteItem.css";


function NoteItem(props){

    const deleteNoteRequest = (noteId, deleteNoteFunction) => {

        fetch(`http://localhost:9090/notes/${noteId}`, {
          method: 'DELETE',
          headers: {
            'content-type': 'application/json'
          }
        })
          .then(res => {
            if (!res.ok) {
              // get the error message from the response,
              return res.json().then(error => {
                // then throw it
                throw error
              })
            }
            return res.json()
          })
          .then(data => {
            // Push main route '/' onto NoteMain's history here once the 
            // API call is successful. This will render the main page instead
            // of trying to render a deleted note!!
            props.history.push('/');
            deleteNoteFunction(noteId);
          })
          .catch(error => {
            console.error(error)
          })
    };

        return (
            <NotefulContext.Consumer>
                {(context) => (
                    <section className="note-item">
                        <h2 className="note-title">
                            <Link to={`/notes/${props.noteId}`}>
                                {props.name}
                            </Link>
                        </h2>
                        <div className="date-delete">
                            <p className="date">{`Date last modified: ${new Date(props.modified)}`}</p>
                            
                            <button
                                className="delete-button"
                                onClick={() => {
                                    deleteNoteRequest(props.noteId, context.deleteNote)}}
                           >
                                Delete
                            </button>
                        </div>
                    </section>
                )}
            </NotefulContext.Consumer>
            
        );  
}

// Use withRouter because it will give us the history from NoteMain thus
// allowing us to push('/') route back into the NoteMain page history.
// This will render the main route '/' instead of trying to render the 
// NoteMain page of a note that we just deleted. 
export default withRouter(NoteItem);

