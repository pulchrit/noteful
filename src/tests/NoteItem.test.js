import React from 'react';
import ReactDOM from 'react-dom';
import {BrowserRouter} from 'react-router-dom';
import toJson from 'enzyme-to-json';
import {shallow} from 'enzyme';
import NoteItem from '../components/NoteItem';

const noteId = 'd26e0570-ffaf-11e8-8eb2-f2801f1b9fd1';
const modified = "2019-01-04T00:00:00.000Z";
const name = "Birds";
const folderId = "b0715efe-ffaf-11e8-8eb2-f2801f1b9fd1";
const content = "Ipsum iusto explicabo sed aut corporis maiores sunt. Qui iure ad quia at sed. Consequ";

describe("NoteItem component", () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <BrowserRouter>
        <NoteItem />
      </BrowserRouter>, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders UI as expected', () => {
    const wrapper = shallow(<NoteItem 
      noteId={noteId}
      modified={modified}
      name={name}
      folderId={folderId}
      content={content} 
      />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
