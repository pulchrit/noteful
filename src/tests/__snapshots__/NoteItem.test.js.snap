// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoteItem component renders UI as expected 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
